src/validation.cpp

```diff
@@ -946,6 +948,16 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
     }
 
     ws.m_ancestors = *ancestors;
+    if (const auto err_string{CheckV3Inheritance(ws.m_ptx, ws.m_ancestors)}) {
```

Comment:
This is an aside -- in the cluster mempool branch, I think I'd like to change this to just look at direct parents rather than ancestors (I believe that in the normal course of transaction validation, we should no longer need to calculate ancestor sets when we're not worried about ancestor limits or updating cached ancestor state anymore).  Can you think of any reason it would be problematic to replace ancestors with just an entry's direct parents here?

Reply:
I think just direct parents should be fine - presumably any further generations should have the same version anyway.
I'll continue using `m_ancestors` here since it's what we have access to, but will leave a comment about this.